pkg/build/webhook/controller.go

```diff
@@ -20,7 +20,7 @@ type Plugin interface {
 	// - newly created build object or nil if default is to be created
 	// - information whether to trigger the build itself
 	// - eventual error.
-	Extract(buildCfg *buildapi.BuildConfig, secret, path string, req *http.Request) (*buildapi.SourceRevision, bool, error)
+	Extract(buildCfg *buildapi.BuildConfig, secret, path string, req *http.Request) (*buildapi.SourceRevision, []kapi.EnvVar, bool, error)
```

Comment:
@PI-Victor i mentioned on scrum this morning that I started looking into it and it's kinda ugly, i think we should just land #8580 as is, and i'll try to finish the refactor i've already started.  i won't merge it until #8580 goes in so you don't have to rebase yet again.